public/video-ui/src/components/FormFields/RichTextEditor.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useRef, useEffect, useState } from 'react';
+import { EditorView } from 'prosemirror-view';
+import { createEditorView } from './richtext/setup';
+import { createSchema } from './richtext/createSchema';
+import { AllSelection, Transaction } from 'prosemirror-state';
+import { DOMParser as parseDOM, Node } from 'prosemirror-model';
+import { EditorConfig } from './richtext/config';
+import { paragraphToWhitespace } from './richtext/utils/richTextHelpers';
+
+
+interface RichTextEditorProps {
+  value: string;
+  onUpdate: (str: string) => void;
+  config: EditorConfig;
+  label?: string;
+  copiedValue?: {
+    text: string,
+    seed: number
+  }
+  shouldAcceptCopiedText: boolean;
+}
+
+export const RichTextEditor = ({ value, onUpdate, config, label, copiedValue, shouldAcceptCopiedText = false }: RichTextEditorProps) => {
+  const schema = createSchema(config);
+
+  const editorEl = useRef<HTMLDivElement>(null);
+  const [editorView, setEditorView] = useState<EditorView | undefined>(
+    undefined
+  );
+
+  useEffect(() => {
+    // Editor view takes an HTML Node therefore this string value needs to be converted into a node by placing in a div
+    const contentNode = document.createElement('div');
+    contentNode.innerHTML = value;
+    const edView = createEditorView(onUpdate, editorEl, contentNode, schema, config);
+    setEditorView(edView);
+  }, []);
+
+  const setContent = (tr: Transaction, node: Node): Transaction => {
+    const sel = new AllSelection(tr.doc);
+    sel.replaceWith(tr, node);
+    return tr;
+  };
+
+  useEffect(() => {
+    if (copiedValue.text && editorView && shouldAcceptCopiedText){
```

Comment:
Is there any reason we couldn't do this with `value` itself?

For example – 

1. User clicks 'copy trail text from description'
2. Button updates underlying model
3. RichTextEditor component rerenders
4. Effect updates ProseMirror instance (we could diff the incoming prop against the current state if we were worried about frequent updates)

This may have already been tried! But keen to avoid the complexity here if poss.

Reply:
I tried and failed - I'd much prefer to do that so it would be good if we could pair.

For some reason anything I tried to update the ProseMirror instance with would leave it empty, except the `pasteHTML` method. I assume there was some subtle error in my implementation but after several hours I ended up with this - it works but it's indirect and messy.